addons/mail/static/src/widgets/form_renderer/form_renderer_tests.js

```diff
@@ -448,6 +457,204 @@ QUnit.test('chatter should become enabled when creation done', async function (a
     );
 });
 
+QUnit.test('Form view not scrolled when switching record', async function (assert) {
+    assert.expect(7);
+
+    this.data['res.partner'].records = [{
+        activity_ids: [],
+        id: 2,
+        display_name: "second partner",
+        description: [...Array(60).keys()].join('\n'),
+        message_ids: [],
+        message_follower_ids: [],
+    }, {
+        activity_ids: [],
+        id: 3,
+        display_name: "third partner",
+        message_ids: [],
+        message_follower_ids: [],
+    }];
+
+    const messages = [...Array(60).keys()].map(id => {
+        let partnerId;
+        if (id % 2 === 0) {
+            partnerId = 2;
+        } else {
+            partnerId = 3;
+        }
+        return {
+            author_id: [10, "Demo User"],
+            body: `<p>Message ${id + 1}</p>`,
+            date: "2019-04-20 10:00:00",
+            id: id + 1,
+            message_type: 'comment',
+            model: 'res.partner',
+            record_name: 'General',
+            res_id: partnerId,
+        };
+    });
+    this.data['mail.message'].records = messages;
+
+    await this.createView({
+        data: this.data,
+        hasView: true,
+        // View params
+        View: FormView,
+        model: 'res.partner',
+        arch: `
+            <form string="Partners">
+                <sheet>
+                    <field name="name"/>
+                    <field name="description"/>
+                </sheet>
+                <div class="oe_chatter">
+                    <field name="message_ids"/>
+                </div>
+            </form>
+        `,
+        viewOptions: {
+            currentId: 2,
+            ids: [2, 3]
+        },
+        config: {
+            device: {size_class: 4}
+        }
+    });
+
+    const controllerContentEl = document.querySelector('.o_content');
+    const formViewEl = document.querySelector('.o_form_view');
+    assert.doesNotHaveClass(formViewEl, 'o_xxl_form_view',
+        "Form view is not xxl"
+    );
```

Comment:
would remove this assertion (useless)

Reply:
nope. The aim is to get rid of the media query present in enterprise. both this assert and the business code ensure we are in the case that we are testing. Since those tests are executed in community and enterprise, we need to be able to differentiate the two. Besides, in some cases we add xss_form_view, it seems relevant to have a xxl_form view (or an algo that put the right class, but this is out of scope i guess)

Reply:
I understand that we want to remove media queries, the less media queries the better...
But I don't think the current solution is good. It adds too much complicated code IMO, and it denies OWL paradigm by mutating the DOM.

Yes, views haven't been converted to OWL yet... but this is not a reason to pile up more bad-looking code (read pretty = maintainable), and since views are planned to being refactored with OWL, I would make changes that goes in this direction, which I feel this change doesn't.

Reply:
Just to make sure this thread is coherent with other threads (see https://github.com/odoo/odoo/pull/53136#discussion_r444118315):
In the end I'm good with replacing media query with JS DOM manipulation with classList, as a less bad solution.

Still doesn't annihilate the 1st comment here to remove this assertion: setup already tells we are in XXL mode, and assertion on such class name is functionally meaningless implementation detail, hence just remove it.